File: 717-1-bit-and-2-bit-characters/1-bit-and-2-bit-characters.cs
public class Solution {
    public bool IsOneBitCharacter(int[] bits) {
        int n = bits.Length;
        int i = 0;

        while (i < n - 1) {
            if (bits[i] == 1) {
                i += 2;
            } else {
                i += 1;
            }
        }

        return i == n - 1;
    }
}

File: 717-1-bit-and-2-bit-characters/1-bit-and-2-bit-characters.py
class Solution:
    def isOneBitCharacter(self, bits: List[int]) -> bool:
        n = len(bits)
        i = 0
        
        while i < n - 1:
            if bits[i] == 1:
                i += 2
            else:
                i += 1
        
        return i == n - 1
